### DISCUSSION TITLE
[Util] Use multiple files, model inheritance, cross-file relations and more!

### DISCUSSION THREAD

**Discussion by DavidHancu** [prisma/prisma#15172] [Q&A] [answered]
# :wave: Hi everyone!

## Introduction

I've been working hard for the past weeks to create a CLI tool that we can all use to make our developer experience better, and the package has just dropped on NPM: [Prisma Util](https://www.npmjs.com/package/prisma-util). The code is also available on GitHub, and you can access it by clicking [here](https://github.com/DavidHancu/prisma-util).

Long gone are the days when we made massive schemas for our databases! With Prisma Util, you can just use multiple files and we'll warn you about naming conflicts too! I also wanted to make sure that this tool integrates seamlessly with your current workflow, and as such I designed it to work as either an addon or a drop-in replacement for the Prisma CLI (I've implemented all of the commands from Prisma and when you use a command from Prisma Util, the package automatically calls the Prisma command with all tweaks done).

## Features

This is a list of all of the current features, including 2 proposals that you can vote on at the end of this message!
─   Multiple schema files for easier and cleaner configuration
─   Model inheritance and mapper for relations
─   Cross-file relations with support for resolving naming conflicts
─   Timing reports and an easy to use CLI
─   Creating schemas using JavaScript (Proposal)
─   Using additional formats (such as YAML) instead of .env files for configuring Prisma (Proposal)

## Proposals
─  [Creating schemas using JavaScript](https://github.com/DavidHancu/prisma-util/issues/1)
─  [Using additional formats instead of .env files for configuring Prisma](https://github.com/DavidHancu/prisma-util/issues/2)

## Final Words
I really hope that this project helps you with some of your problems or short-comings of using Prisma and makes the developer experience at least a bit better. If you have any suggestions, please don't hesitate to create an issue or respond here, as well as vote on the proposals above. Thank you for reading this!

**Comment by austincrim** [comment, marked as answer]
Hey David, this is a super cool project, great work!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
